iceoryx_hoofs/testing/mocks/error_handler_mock.cpp

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "iceoryx_hoofs/testing/mocks/error_handler_mock.hpp"
+
+namespace iox
+{
+std::mutex ErrorHandlerMock::handler_mutex;
```

Comment:
Does this snake case break with our (local) conventions or is there more of a purpose?

I prefer snake case for certain library abstractions but not variables like this one, unless all variables in the module follow this convention.

But more importantly, it is a global variable in the iox namespace but is actually something specific to error handling. We should try to avoid polluting the iox namespace with this. If we cannot avoid the global mutex (can we confine it to some class as a static?) maybe it should live in something like `iox::error_handling::detail`.

The convention being that the any `detail` namespace is certainly not to be accessed by the user (even if it can be accessed to write a global variable like this one).

Reply:
Fixed the snake case naming. We can think about wrapping parts of the `error_handling` in objects in an extra namespace. For me this not an important topic, as the class is already separate in a testing package. Hence, not a big deal for me.